### PR TITLE
fix: show `-` for empty data in run table

### PR DESCRIPTION
**Pull request by keita-determined**
## Ticket

[ET-80]
Related PR: https://github.com/determined-ai/hew/pull/135

## Description

Empty entries in the run table should contain a dash

## Test Plan

- If the cell value is empty (null value), show the dash `-` in the run table

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-80]: https://hpe-aiatscale.atlassian.net/browse/ET-80?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Screenshot 2024-08-30 at 09-37-56 Uncategorized Runs - Determined
- Forked column doesn't have dashes
- I think it would look better if the dashes were centered in the column, if possible
- It doesn't make much sense to allow users to right click and copy the "value" of `-` for an empty column. That's confusing behavior and it would be better if the dash was display only.

**Comment by keita-determined** [comment]
> Forked column doesn't have dashes

fixed 
Note: ideally, empty `fork` cell should be shown as text, but  it's difficult to change the cell type when the data is empty. Even if there's no fork data, empty fork data, `-`, is shown as a link. The link doesn't go anywhere, and cursor icon is not a hand icon. 

https://github.com/user-attachments/assets/d0cc2d67-b8e9-4ace-9d57-d11a0db068b2


> I think it would look better if the dashes were centered in the column, if possible

it's difficult to make them centered.

> It doesn't make much sense to allow users to right click and copy the "value" of - for an empty column. That's confusing behavior and it would be better if the dash was display only.

fixed. removed copy button if data is empty.

**Comment by EmilyBonar** [comment]
> > Forked column doesn't have dashes
> 
> fixed Note: ideally, empty `fork` cell should be shown as text, but it's difficult to change the cell type when the data is empty. Even if there's no fork data, empty fork data, `-`, is shown as a link. The link doesn't go anywhere, and cursor icon is not a hand icon.

> > I think it would look better if the dashes were centered in the column, if possible
> 
> it's difficult to make them centered.

Couldn't you create a `EMPTY_CELL` constant that looked something like:
```
{
      allowOverlay: false,
      data: '-',
      displayData: '-',
      kind: GridCellKind.Text,
      contentAlign: 'center'
}
```
And then you could just the following, which should fix both of the above problems:
```
renderer: (record: FlatRun) => (record.checkpointSize ? 
    {
      allowOverlay: false,
      copyData: humanReadableBytes(record.checkpointSize),
      data: { kind: TEXT_CELL },
      kind: GridCellKind.Custom,
    } : 
    EMPTY_CELL),
```

**Comment by keita-determined** [comment]
@EmilyBonar thats a good idea. made some changes. 
not sure if we wanna do `contentAlign: 'center'` though. w&b and some others follows the column alignment. in our case, left aligned is the default. Also, centered dash might be harder to look when column width is long. I think we should ask product people first if we wanna make that move anyways.

File: webui/react/src/pages/FlatRuns/columns.ts [hunk 1]
```diff
@@ -99,6 +104,13 @@ function getCellStateFromExperimentState(expState: RunState) {
   }
 }
 
+const EMPTY_CELL: GridCell = {
+  allowOverlay: false,
+  copyData: '-',
+  data: { kind: TEXT_CELL },
+  kind: GridCellKind.Custom,
```
Comment (EmilyBonar) [inline webui/react/src/pages/FlatRuns/columns.ts:111]: nit: why should this be a custom cell type instead of just text?

**Review by EmilyBonar — APPROVED** [review]
I feel like an empty column marker being centered is fairly standard, but I suppose it doesn't matter. This looks good, but should it be applied to the Searches table as well?

File: webui/react/src/pages/FlatRuns/columns.ts (hunk 1 above)
Comment (keita-determined) [reply webui/react/src/pages/FlatRuns/columns.ts:111]: That causes type error with `GridCellKind.TEXT`.

**Comment by keita-determined** [comment]
> This looks good, but should it be applied to the Searches table as well?

I think so. The ticket specifies run table, and i don't have enough capacities now. Made a ticket https://hpe-aiatscale.atlassian.net/browse/ET-749, so hope someone else can work on it.

File: webui/react/src/pages/FlatRuns/columns.ts (hunk 1 above)
Comment (EmilyBonar) [reply webui/react/src/pages/FlatRuns/columns.ts:111]: It's merged now so I guess it's too late, but this doesn't give me a type error:
```
const EMPTY_CELL: GridCell = {
  allowOverlay: false,
  data: '-',
  kind: GridCellKind.Text,
};
```

Comment (keita-determined) [reply webui/react/src/pages/FlatRuns/columns.ts:111]: image

looks like this works. i think we can change it with searches table changes

```ts
const EMPTY_CELL: GridCell = {
  allowOverlay: false,
  data: '-',
  displayData: '-',
  kind: GridCellKind.Text,
} as const;
```

**Pull request merged**